CHANGES.md

```diff
@@ -1,9 +1,44 @@
-Unreleased
-----------
+3.18.0~alpha0 (2025-03-32)
+--------------------------
 
-If you're a contributor, please include your CHANGES entry in a file
-`doc/changes/$PR_NAME.md`. At release time, it will be incoporated into the
-changelog properly.
+### Fixed
+
+- Support HaikuOS: don't call `execve` since it's not allowed if other pthreads
+  have been created. The fact that Haiku can't call `execve` from other threads
+  than the principal thread of a process (a team in haiku jargon), is a
+  discrepancy to POSIX and hence there is a [bug about
+  it](https://dev.haiku-os.org/ticket/18665). (@Sylvain78, #10953)
+- Fix flag ordering in generated Merlin configurations (#11503, @voodoos, fixes
+  ocaml/merlin#1900, reported by @vouillon)
+- Display negative error codes on Windows in hex which is the more customary
+  way to display `NTSTATUS` codes (#11504, @MisterDA)
+
+### Added
+
+- Add `(format-dune-file <src> <dst>)` action.  (#11166, @nojb)
```

Comment:
I think it would be nice to extend why this is useful and allows to avoid calling the internal dune formatting command but it depends how much context we want to give in the changelog.